views/default/resources/friend_request.php

```diff
@@ -1,12 +1,17 @@
 <?php
 
-elgg_gatekeeper();
-
-$username = elgg_extract('username', $vars);
-$user = get_user_by_username($username);
+$request = elgg_extract('request', $vars);
+/* @var $request \Elgg\Request */
+
+$username = $request->getParam('username');
+if ($username) {
```

Comment:
since `username` is a requirement of the route there is no use case where this is empty